src/types/app-client.ts

```diff
@@ -2162,24 +2160,31 @@ export class ApplicationClient {
       call?.method &&
       // We aren't skipping the send
       !call.sendParams?.skipSending &&
-      // There isn't an ATC passed in
-      !call.sendParams?.atc &&
+      // There isn't an composer passed in
+      !call.sendParams?.transactionComposer &&
       // The method is readonly
-      this.appSpec.hints[this.getABIMethodSignature(this.getABIMethod(call.method)!)].read_only
+      this.appSpec.hints?.[this.getABIMethodSignature(this.getABIMethod(call.method)!)]?.read_only
     ) {
-      const atc = new AtomicTransactionComposer()
-      await this.callOfType({ ...call, sendParams: { ...call.sendParams, atc } }, 'no_op')
-      const result = await atc.simulate(this.algod)
+      const transactionComposer = new TransactionComposer({
+        algod: this.algod,
+        getSigner: (address) => {
```

Comment:
We do this pattern in a few places, so we could extract this function and share across the code.